### ISSUE TITLE
VIZIO Chromium Failure on Seeking Backwards

### ISSUE THREAD

**Issue by jamescahall** [shaka-project/shaka-player#2463]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**What version of Shaka Player are you using?**
2.5.9

**Can you reproduce the issue with our latest release version?**
Yes

**Can you reproduce the issue with the latest code from `master`?**
Yes

**Are you using the demo app or your own custom app?**
Custom

**If custom app, can you reproduce the issue using our demo app?**
Unable to test on VIZIO TV

**What browser and OS are you using?**
VIZIO Chromium

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**
VIZIO E55-E1 - 9.0.5.2

**What are the manifest and license server URIs?**
DRM Not enabled in test

**What did you do?**
Loaded player with start time > 0
            this.drmPlayer.load(videoUrl, startTime).then(function() {
                _this.play();
            }).catch(function(error) {
                log("DRM Load Error: " + error);
            });

Seek backwards after playback starts:
                _this.drmPlayer.getMediaElement().currentTime = seekTime;

Seeking forward works fine

**What did you expect to happen?**
Player should seek to desired time

**What actually happened?**
Playback Fails with Error:
The video playback was aborted due to a corruption problem or because the video used features your browser did not support.

Error Code 3016
"CHUNK_DEMUXER_ERROR_APPEND_FAILED: Parsed buffers not in DTS sequence"

Followed by Error Code 3014

**Comment by jamescahall** [comment]
Additional Data:
        configuration = { 'abr': { 'defaultBandwidthEstimate' : 500000 }, 'streaming': { 'bufferingGoal' : 10, 'rebufferingGoal': 4 } };

I also tested in Chrome 80 (not on VIZIO TV) and can seek backwards, but when I seek backwards behind startTime there is no audio until I get to startTime again.

**Comment by TheModMaker** [comment]
I believe we don't have a VIZIO device, and even if we did, COVID has restricted our access to our lab.  So we aren't able to test this ourselves.  This is why we added the `help wanted` tag.  Without access to a device, we have no way to test and fix any issues.  If you have any suggestions to fix, we'd be happy to review a PR.

**Comment by MilosRasic** [comment]
> If custom app, can you reproduce the issue using our demo app?
> Unable to test on VIZIO TV

Maybe try to hardcode a demo stream manifest URL in your app and check if that works fine on the TV. That could show whether the problem is triggered by your code or your content. Is it DASH or HLS you're trying to play?

Also, are you using the new casting tool that asks you to pair with the device? If unsure, which Chromium version is the TV reporting?

**Comment by joeyparrish** [comment]
The message "Parsed buffers not in DTS sequence" might be a useful clue.  Could it be that the content isn't compatible with the decoder in some way?

But as @TheModMaker said, we don't have such a device and can't officially support platforms on which we can't test or validate our software.  We are happy to have contributions from the community and merge code to support additional platforms when we can, but we would be making blind guesses if we tried to develop those patches ourselves.

**Comment by jamescahall** [comment]
The latest VIZIO TVs from 2019+ use the Conjure Chromium browser and no longer uses the Google Cast Browser.  This is HLS video and plays fine and can seek forward but unable to seek backwards without failure.  Seeking works on Chrome using same code but is slow to buffer going backwards.  For seek, we are simply setting the currentTime value on the media element during playback (no pause, etc.).  I can provide an example video file by email if need be.

**Comment by avelad** [comment]
Can you test with v4.3.4? Thanks!

**Comment by jamescahall** [comment]
Unfortunately we are no longer using Shaka on VIZIO due to this issue taking too long to resolve and using VideoJS on that platform.

**Comment by avelad** [comment]
Sorry about that...

**Comment by joeyparrish** [comment]
@jamescahall [commented](https://github.com/shaka-project/shaka-player/issues/2463#issuecomment-1410844278):
> this issue taking too long to resolve

@joeyparrish [commented](https://github.com/shaka-project/shaka-player/issues/2463#issuecomment-661253892) 2.5 years earlier:
> we don't have such a device and can't officially support platforms on which we can't test or validate our software

Sorry if we weren't clear, but working on bugs on unsupported platforms is the responsibility of those in the community with the devices.  There's no way we can fix Vizio issues without a Vizio and a way to integrate it with our test infra.

**Comment by jamescahall** [comment]
The lack of having a test device seems to be common issue across many of the smart TVs.  We have connections at most of these companies.  Is there something that is blocking you getting these devices?  We can potentially help you get free devices if that is an issue.

**Comment by joeyparrish** [comment]
We need a device and a connection to our Selenium grid.  We've been building out [Generic WebDriver Server](https://github.com/shaka-project/generic-webdriver-server) to expand the Selenium grid.  Currently, it supports ChromeCast, Tizen, and Xbox One.  LG WebOS support is a WIP, but we have an LG device in the lab ready to go.

**Comment by jamescahall** [comment]
Ok.  I'd like to get a chance to chat with you more 1 on 1 on this as we've had to use a mix of ShakaPlayer, VideoJS, AVPlay, Native HTML5 video tag with extensions, etc. to fully support the 50+ brands of TVs we launch apps on.  Getting the full combo of Clear Video, DRM Video, MPEG-DASH, HLS, In-manfest captions, Embedded captions, VOD, and Linear (with discontinuities) to work on various platforms has been extremely painful over the last 10+ years.

We've only been able to use Shaka on:
Xbox
LG (still with some stability issues with Linear playback)
Comcast / Cox

For linear specifically, we've had trouble figuring out the best combinations of settings to get them to work well and we can't share the stream urls in these tickets.

The rest of the 50+ platforms we have to use other players.
